lib/BoundaryValueDiffEqShooting/src/multiple_shooting.jl

```diff
@@ -209,15 +217,27 @@ function __multiple_shooting_init_odecache(
 end
 
 function __multiple_shooting_init_jacobian_odecache(
-        ensemblealg, prob, jac_cache, ::NoDiffCacheNeeded, alg, nshoots, u; kwargs...)
+        ensemblealg, prob, jac_cache, diffmode, alg, nshoots, u; kwargs...)
+    __multiple_shooting_init_jacobian_odecache(
+        ensemblealg, prob, jac_cache, __cache_trait(diffmode),
+        diffmode, alg, nshoots, u; kwargs...)
+end
+
+function __multiple_shooting_init_jacobian_odecache(
+        ensemblealg, prob, jac_cache, ::NoDiffCacheNeeded,
+        diffmode, alg, nshoots, u; kwargs...)
     return __multiple_shooting_init_odecache(ensemblealg, prob, alg, u, nshoots; kwargs...)
 end
 
 function __multiple_shooting_init_jacobian_odecache(
-        ensemblealg, prob, jac_cache, ::DiffCacheNeeded, alg, nshoots, u; kwargs...)
-    cache = jac_cache.cache
-    if cache isa ForwardDiff.JacobianConfig
-        xduals = reshape(cache.duals[2][1:length(u)], size(u))
+        ensemblealg, prob, jac_cache, ::DiffCacheNeeded,
+        diffmode, alg, nshoots, u; kwargs...)
+    if diffmode isa AutoSparse
+        T_dual = eltype(overloaded_input_type(jac_cache))
+        xduals = zeros(T_dual, size(u))
+    elseif diffmode isa AutoForwardDiff
+        T_dual = eltype(overloaded_input_type(jac_cache))
+        xduals = zeros(T_dual, size(u))
     else
         xduals = reshape(cache.t[1:length(u)], size(u))
```

Comment:
Is this line tested? It seems like it would error with the new code, and also it relies on internals

Reply:
Actually this line can be safely removed since this dispatch is specifically designed for cases using ForwardDiff